### PR TITLE
feat: health indicator for Prisma ORM

### PR DESCRIPTION
**Pull request by ThallesP**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/terminus/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Feature

## What is the current behavior?

Issue Number: #1510 

## What is the new behavior?
Add supports for the Prisma ORM

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
~~Currently it only supports SQL based databases, I'm still working on the MongoDB integration.~~

### CODE REVIEW

**Comment by BrunnerLivio** [comment]
Hey @ThallesP appreciate this PR. The baseline looks good so far. Would it be possible to fix the CI issues as well as adding an e2e test?

**Comment by ThallesP** [comment]
> Hey @ThallesP appreciate this PR. The baseline looks good so far. Would it be possible to fix the CI issues as well as adding an e2e test? For the e2e test you can inspire yourself from the [mikro orm test](https://github.com/nestjs/terminus/blob/3c2c2c3c8da6870673d4cd6bbb9792efcf9bb86f/e2e/health-checks/mikro-orm.health.e2e-spec.ts)

Yes, no problem, I'm still finishing it (I should've marked this PR as a draft).

**Comment by ThallesP** [comment]
Ready for review, added tests too @BrunnerLivio

File: sample/012-prisma-app/package.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "jest --config ./test/jest-e2e.json"
+  },
+  "dependencies": {
+    "@nestjs/common": "9.3.2",
+    "@nestjs/core": "9.3.2",
+    "@nestjs/platform-express": "9.3.2",
+    "@nestjs/terminus": "9.2.2",
```
Comment (ThallesP) [inline sample/012-prisma-app/package.json:26]: this should be updated to a newer version with the Prisma changes or else it will fail.

**Review by ThallesP — COMMENTED** [review]
the prisms sample should be updated to a newer version with these PR changes.

**Comment by ThallesP** [comment]
@BrunnerLivio prisma only supports v14 and above, might need to bump the node version.
https://www.prisma.io/docs/reference/system-requirements

**Comment by BrunnerLivio** [comment]
I merged this PR in the `next` branch for now, since I can't publish it on the latest distribution because I can't drop support for Node v12 yet.
So this will probably land in Terminus v10. Will continue the discussion over at #1510

**Comment by ThallesP** [comment]
@BrunnerLivio technically we don't depend on `@prisma/client`, if you see I just create an interface that represents the prisma client, so it isn't needed to drop support for node v12, you would need v14+ if you're using the prisma health indicator.

but the ci/cd would be broken as it tests against node v12.

**Comment by BrunnerLivio** [comment]
@ThallesP I am aware it _technically works_ but I can't drop the Node v12 Pipeline just for Prisma. It's for sure technically possible to not install Prisma or run the Prisma-related tests during the v12 Pipeline but I'd prefer not to jump through these hoops.
[Nest v10 is anyway on it's way ](https://github.com/nestjs/nest/tree/10.0.0) so I prefer to save myself from any short-lived time investment

**Pull request merged**